File: libraries/core_legacy/src/main/java/com/tkpd/library/utils/legacy/SessionAnalytics.java
package com.tkpd.library.utils.legacy;

import com.tokopedia.track.TrackApp;

/**
 * Created by Yoris Prayogo on 22/12/20.
 * Copyright (c) 2020 PT. Tokopedia All rights reserved.
 */
public class SessionAnalytics {

    private static final String EVENT = "clickLogout";
    private static final String EVENT_CATEGORY = "force logout";
    private static final String EVENT_ACTION = "refresh token";

    public static void trackRefreshTokenSuccess(){
        TrackApp.getInstance().getGTM().sendGeneralEvent(
                EVENT,
                EVENT_CATEGORY,
                EVENT_ACTION,
                "refresh token success"
        );
    }

    public static void trackRefreshTokenFailed(String errorMsg){
        TrackApp.getInstance().getGTM().sendGeneralEvent(
                EVENT,
                EVENT_CATEGORY,
                EVENT_ACTION,
                "refresh token failed - " + errorMsg
        );
    }
}

File: libraries/core_legacy/src/main/java/com/tkpd/library/utils/legacy/SessionAnalytics.kt
package com.tkpd.library.utils.legacy

import com.tokopedia.track.TrackApp

/**
 * Created by Yoris Prayogo on 21/12/20.
 * Copyright (c) 2020 PT. Tokopedia All rights reserved.
 */

class SessionAnalytics {
    companion object {
        const val EVENT = "clickLogout"
        const val EVENT_CATEGORY = "force logout"
        const val EVENT_ACTION = "refresh token"

        fun trackRefreshTokenSuccess(){
            TrackApp.getInstance().getGTM().sendGeneralEvent(
                    EVENT,
                    EVENT_CATEGORY,
                    EVENT_ACTION,
                    "refresh token success"
            )
        }

        fun trackRefreshTokenFailed(errorMsg: String){
            TrackApp.getInstance().getGTM().sendGeneralEvent(
                    EVENT,
                    EVENT_CATEGORY,
                    EVENT_ACTION,
                    "refresh token failed - $errorMsg"
            )
        }
    }
}